grease/src/Grease/SymIO.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ImplicitParams #-}
+
+-- |
+-- Copyright        : (c) Galois, Inc. 2025
+-- Maintainer       : GREASE Maintainers <grease@galois.com>
+module Grease.SymIO (
+  InitializedFs (..),
+  initialLlvmFileSystem,
+) where
+
+import Control.Monad qualified as Monad
+import Data.Map.Strict qualified as Map
+import Data.Parameterized.NatRepr (knownNat)
+import Data.Text qualified as Text
+import Grease.Options qualified as GO
+import Lang.Crucible.Backend qualified as C
+import Lang.Crucible.FunctionHandle qualified as C
+import Lang.Crucible.LLVM.MemModel qualified as Mem
+import Lang.Crucible.LLVM.SymIO qualified as CLLVM.SymIO
+import Lang.Crucible.Simulator qualified as C
+import Lang.Crucible.SymIO qualified as SymIO
+import Lang.Crucible.SymIO.Loader qualified as SymIO.Loader
+import What4.Interface qualified as W4
+
+data InitializedFs sym ptrW
+  = InitializedFs
+  { initFsContents :: SymIO.InitialFileSystemContents sym
+  , initFs :: CLLVM.SymIO.LLVMFileSystem ptrW
+  , initFsGlobals :: C.SymGlobalState sym
+  , initFsOverride :: CLLVM.SymIO.SomeOverrideSim sym ()
+  }
+
+-- | Initialize the symbolic file system.
+initialLlvmFileSystem ::
+  ( C.IsSymInterface sym
+  , Mem.HasPtrWidth ptrW
+  ) =>
+  C.HandleAllocator ->
+  sym ->
+  GO.FsOpts ->
+  IO (InitializedFs sym ptrW)
+initialLlvmFileSystem halloc sym simOpts = do
```

Comment:
Perhaps this should be named "`fsOpts`" instead of "`simOpts`s", given that it's of type `FsOpts` instead of `SimOpts`.